store/confirmCDS.go

```diff
@@ -83,57 +85,70 @@ func (m mongoDB) GetCDSConfirm(loc schema.Location) (float64, float64, float64,
 	case CdsTaiwan:
-		// use taiwan cdc data (temp solution)
-		today, delta, percent, err := m.GetConfirm(loc)
-		if err != nil {
-			return 0, 0, 0, fmt.Errorf("%s: %s", ErrConfirmDataFetch, err)
+		c = m.client.Database(m.database).Collection(CDSCountyCollectionMatrix[CDSCountryType(CdsTaiwan)])
+		opts := options.Find().SetSort(bson.M{"report_ts": -1}).SetLimit(2)
+		filter := bson.M{}
+		cur, err = c.Find(context.Background(), filter, opts)
+		if nil != err {
+			log.WithField("prefix", mongoLogPrefix).Errorf("CDS confirm data find  error: %s", err)
+			return 0, 0, 0, fmt.Errorf("cds confirm data find  error: %s", err)
 		}
-		return today, delta, percent, nil
 
 	case CdsUSA:
-		c := m.client.Database(m.database).Collection(CDSCountyCollectionMatrix[CDSCountryType(CdsUSA)])
-		ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
-		defer cancel()
-		opts := options.Find()
-		opts = opts.SetSort(bson.M{"report_ts": -1}).SetLimit(2)
+		c = m.client.Database(m.database).Collection(CDSCountyCollectionMatrix[CDSCountryType(CdsUSA)])
+		opts := options.Find().SetSort(bson.M{"report_ts": -1}).SetLimit(2)
 		filter := bson.M{"county": pGeo.Level2, "state": pGeo.Level1}
-		cur, err := c.Find(context.Background(), filter, opts)
+		cur, err = c.Find(context.Background(), filter, opts)
 		if nil != err {
 			return 0, 0, 0, ErrConfirmDataFetch
 		}
-		var results []schema.CDSData
+	default:
+		return 0, 0, 0, ErrNoConfirmDataset
+	}
 
-		for cur.Next(ctx) {
-			var result schema.CDSData
-			if errDecode := cur.Decode(&result); errDecode != nil {
-				return 0, 0, 0, ErrConfirmDecode
-			}
-			results = append(results, result)
-		}
-		percent := float64(100)
-		if len(results) >= 2 {
-			if results[0].ReportTime > results[1].ReportTime {
-				today := results[0]
-				yesterday := results[1]
-				delta := today.Cases - yesterday.Cases
-				if yesterday.Cases > 0 {
-					percent = 100 * delta / yesterday.Cases
-				}
-				log.WithField("prefix", mongoLogPrefix).Debugf("cds score results: today:%f, delta:%f, percent:%f", today.Cases, delta, percent)
-				return today.Cases, delta, percent, nil
-			} else if 1 == len(results) {
-				today := results[0]
-				delta := today.Cases
-				log.WithField("prefix", mongoLogPrefix).Debugf("cds score results: today:%f, delta:%f, percent:%f", today.Cases, delta, percent)
-				return today.Cases, delta, percent, nil
-			}
-			return 0, 0, 0, ErrInvalidConfirmDataset
+	for cur.Next(ctx) {
+		var result schema.CDSData
+		if errDecode := cur.Decode(&result); errDecode != nil {
+			return 0, 0, 0, ErrConfirmDecode
 		}
+		log.WithField("prefix", mongoLogPrefix).Debugf("cds query name: %s date:%s", result.Name, result.ReportTimeDate)
+		results = append(results, result)
 	}
-	return 0, 0, 0, ErrNoConfirmDataset
 
+	var percent, delta float64
+	var today, yesterday schema.CDSData
+	if len(results) >= 2 {
+		if results[0].ReportTime > results[1].ReportTime {
```

Comment:
The query result is already sorted by `report_ts`, why is the time comparison needed here?

For the percent rate calculation, the util function is available.
https://github.com/bitmark-inc/autonomy-api/blob/master/score/utils.go#L3

Reply:
They can be equal

Reply:
it can be equal